### PR TITLE
Added button to select attributes from largest geometry when merging features

### PR DESCRIPTION
**Pull request by uclaros**
## Description
Inspired by #42194, I didn't want to make this automatic since there is some logic for fields with default values present.
image
Only enabled for Polygon and Line layers.

Also changed the *autoDefault* property for the first three buttons so that they do not capture the *enter* key and allow it to accept the dialog.

### CODE REVIEW

File: src/app/qgsmergeattributesdialog.cpp [hunk 1]
```diff
@@ -460,21 +484,49 @@ void QgsMergeAttributesDialog::mFromSelectedPushButton_clicked()
...
+  switch ( mVectorLayer->geometryType() )
+  {
+    case QgsWkbTypes::LineGeometry:
     {
-      currentComboBox->setCurrentIndex( currentComboBox->findData( QStringLiteral( "skip" ) ) );
+      QgsFeatureList::const_iterator f_it = mFeatureList.constBegin();
+      for ( ; f_it != mFeatureList.constEnd(); ++f_it )
+      {
+        double featureLength = f_it->geometry().length();
+        if ( featureLength > maxValue )
```
Comment (nyalldawson) [inline src/app/qgsmergeattributesdialog.cpp:503]: if all lengths are zero then featureId will never be set. Possibly a dumb corner case, but maybe this should be >= or maxValue should start at -1 ?

File: src/ui/qgsmergeattributesdialogbase.ui [hunk 2]
```diff
@@ -57,7 +60,53 @@
...
+       </property>
+       <property name="autoDefault">
+        <bool>false</bool>
+       </property>
+      </widget>
+     </item>
+     <item>
+      <widget class="QLabel" name="mTakeLargestAttributesLabel">
+       <property name="text">
+        <string>Take attributes from largest geometry</string>
```
Comment (nyalldawson) [inline src/ui/qgsmergeattributesdialogbase.ui:87]:
```suggestion
        <string>Take attributes from feature with the largest geometry</string>
```

**Comment by nyalldawson** [comment]
Nice feature!

Regarding the icon -- I'd suggest either:
1 . A shaded with polygon with a ">" symbol overlaid
2. Having two polygons, one clearly larger than the other and having the larger polygon appear in the yellow selected color

File: src/app/qgsmergeattributesdialog.cpp (hunk 1 above)
Comment (uclaros) [reply src/app/qgsmergeattributesdialog.cpp:503]: This is kind of on purpose. There is no reason to select one featureId randomly when there is not one larger geometry, and there is line 528 to safeguard uninitialized use of the featureId.

**Comment by uclaros** [comment]
> Regarding the icon -- I'd suggest either:
> 1 . A shaded polygon with a ">" symbol overlaid
> 2. Having two polygons, one clearly larger than the other and having the larger polygon appear in the yellow selected color

I'll see what I can come up with!

Maybe I could also support Multi-Points too and take the geometry with the most parts?

File: src/app/qgsmergeattributesdialog.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/qgsmergeattributesdialog.cpp:503]: Ah - gotcha. All good then!

**Comment by nyalldawson** [comment]
> I'll see what I can come up with!

If you're not feeling inspired then just reuse an existing icon instead of adding a new one (pick any old random one :P ) and open a ticket requesting a new icon, and I'll add it to https://github.com/qgis/QGIS/projects/1

**Comment by uclaros** [comment]
I updated the icon and added support for selecting multipoints based on partCount.
image

The appropriate row is now also selected so it gets highlighted for visual confirmation.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**